apps/mobile/src/components/ContextMenu/ContextMenu.js

```diff
@@ -0,0 +1,217 @@
+          </TouchableOpacity>
+        </View>
+      )}
+    </View>
+  )
+}
+
+function ContextMenuItem ({ widget, groupSlug, rootPath }) {
+  const { t } = useTranslation()
+  const { listItems, loading } = useGetContextWidgetChildren({ widget, groupSlug })
+  const logout = useLogout()
+  const hasResponsibility = useHasResponsibility({ forCurrentGroup: true, forCurrentUser: true })
+  const canAdmin = hasResponsibility(RESP_ADMINISTRATION)
+  const [{ currentGroup }] = useCurrentGroup()
+
+  const title = widget.title
+  const url = makeWidgetUrl({ widget, rootPath, groupSlug })
+
+  const handleWidgetPress = widget => {
+    const context = currentGroup?.isContextGroup ? currentGroup?.slug : 'groups'
+    const linkingPath = makeWidgetUrl({ widget, rootPath, groupSlug: currentGroup?.isContextGroup ? null : currentGroup?.slug })
+
+    openURL(linkingPath)
+  }
+  if (widget.isHiddenInContextMenu) return null
+  if (widget.visibility === 'admin' && !canAdmin) return null
+
+  if (widget.type === 'logout') {
+    return (
+      <TouchableOpacity
+        onPress={() => logout()}
+        className='flex-row items-center p-3 bg-background border-2 border-foreground/20 rounded-md mb-2 gap-2'
+      >
+        <WidgetIconResolver widget={widget} className='mr-2' />
+        <Text className='text-sm font-bold text-foreground'>{title}</Text>
+      </TouchableOpacity>
+    )
+  }
+
+  if (url && (widget.childWidgets.length === 0 && !['members', 'about'].includes(widget.type))) {
+    return (
+      <TouchableOpacity
+        onPress={() => handleWidgetPress(widget)}
+        className='flex-row items-center p-3 bg-background border-2 border-foreground/20 rounded-md mb-2 gap-2'
+      >
+        <WidgetIconResolver widget={widget} className='mr-2' />
+        <Text className='text-sm font-bold text-foreground'>{title}</Text>
+      </TouchableOpacity>
+    )
+  }
+
+  return (
+    <View className='border-2 border-foreground/20 rounded-md p-2 bg-background text-foreground mb-[.5rem]'>
+      {widget.view && (
+        <TouchableOpacity
+          onPress={() => handleWidgetPress(widget)}
+          className='flex-row justify-between items-center content-center'
+        >
+          <Text className='text-sm font-semibold text-foreground'>{title}</Text>
+        </TouchableOpacity>
+      )}
+      {!widget.view && (
+        <View className='flex-row justify-between items-center content-center'>
+          <Text className='text-sm font-semibold text-foreground'>{title}</Text>
+        </View>
+      )}
+      <View className='flex flex-col justify-center items-center relative'>
+        {/* Need to add the SpecialTopElementRenderer here */}
+      </View>
+      {loading && <Text>{t('Loading...')}</Text>}
+      {listItems.length > 0 && listItems.map(item => <ListItemRenderer key={item.id} item={item} rootPath={rootPath} groupSlug={groupSlug} handleWidgetPress={handleWidgetPress} />)}
```

Comment:
Please put a component on a new line and prop all on separate lines when it is s a long list (like ~ over 120 characters wide). This will be challenged in several cases, like I would put the Loading... component above also between parens and on a new line, but totally won't complain if you don't. Long lines like this though I have fairly consistently in the Mobile code base put on their own line, and long props lists get new lines. I'd like to keep doing that:

```
listItems.length > 0 && listItems.map(item => (
  <ListItemRenderer
      key={item.id}
      item={item}
      rootPath={rootPath}
      groupSlug={groupSlug}
      handleWidgetPress={handleWidgetPress}
    />
  ))}